### DISCUSSION TITLE
Remote generation API

### DISCUSSION THREAD

**Discussion by treeder** [zeromintapp/community#5] [General]
**This feature is still in beta, please let us know if you'd like to try it**

If you'd like to generate random NFTs or have some kind of blind mint process, Zeromint has an easy way to do this without having to write complicated smart contracts. https://stupidzombies.club/ is a live example of this. 

All that is required is an API endpoint that returns the following information:

```json
{
    "nft": {
            "name": "Something #X",
            "description": "your description",
            "image": "a URL to an image or video"
    }
}
```

As part of the request, you'll receive and can optionally use:

```json
{ "copyNum": 123 }
```

Once you have that ready, create an NFT on Zeromint, and set the number of copies, eg: 10000. 

Then contact support with your API URL and we'll help you continue. Once this is generally available, you won't need to contact support.

**Comment by wwitkdqsck** [comment]
Hello Treeder,Sorry Im new to this,how do i create an API end point ? Do i need to do any Pull Request ?

**Reply by treeder** [reply]
You'd create your own API and host it somewhere. If you don't know how to do this, you'd probably want to talk to someone who can help you.

**Reply by wwitkdqsck** [reply]
Okay 👌 noted,before I search on the Web for answers,by api you mean a. Json file? And can I use free api hosting services? Thanks!

**Reply by treeder** [reply]
You can use whatever you want to host it. It would have to return different data for every NFT though, like a new image at least, so it can't just be a static JSON file. A serverless function service would work fine, like firebase functions or cloudflare workers.

**Reply by wwitkdqsck** [reply]
okay thanks man ! Can i use ipfs url for the images or only web2.0 ?

**Reply by wwitkdqsck** [reply]
First i thought ,its just a bulk import images on zeromint website itself ,instead of importing a file ,users can chose to Bulk Import or Import 1 File,but looks like it's more complex than i thought. Ill try and do it ! Thanks For the Guidance !

**Reply by treeder** [reply]
Bulk import is another option we're working on, but then it's not blind. If you'd like to try bulk import, let me know. Would require a JSON file with an array of each nft in the same format as above.

**Reply by wwitkdqsck** [reply]
By Blind you mean Users/Buyers can't see what they are getting?Yes would like to test the Bulk Import Method Out..ill attach a .JSON file here soon ! Thanks Travis !

**Comment by wwitkdqsck** [comment]
okay i've created 10 nft collection ,attached is the .JSON file ,Next i need to improve the .JSON file for 1000 nfts by looping the directory and setting all the information/attributes....[CID using GOFS](https://gofs.io/status/bafkreig2zke7hr7nnjsi3qslo6uhecxiqf3rt3bwdb6wvwcx2m4354wpri)
[10 NFTs Collection On Zeromint ](https://zeromint.com/nfts/2vmo1Z8z-eJrKwl2BtW_k) Btw,how do i set it as 10 Free NFTS ?

**Reply by wwitkdqsck** [reply]
hello guys @treeder @r-gochain  
Can this work ? by fetch method ?
https://wwitkdqsck.github.io/api/data.json
https://zeromint.com/nfts/mM4RlEjx0ZE0fRnaH7nxz

By The Way ,Whats the process for New NFT Creators On Zeromint  to Create a Blind Minting Like Stupid Zombies Club If they doesnt know anything about API (They only provide all the images to the collection)  ?Email Zeromint Support Team ? and What will be the charges like?For this Special Features?

**Reply by treeder** [reply]
I'd recommend using something like [Firebase Functions](https://firebase.google.com/docs/functions/get-started) or [Cloudflare workers](https://workers.cloudflare.com/) , they are perfect for this type of thing.

**Reply by wwitkdqsck** [reply]
thanks for the input @treeder , btw is it an API to GET and POST ?

**Reply by treeder** [reply]
just get

**Reply by wwitkdqsck** [reply]
hello @treeder ,@r-gochain  ,thanks for pointing out Firebase Functions appreaciate it....Is this what ur looking for ?
here is the API endpoint



MockAPI

https://61f4e42262f1e300173c3f3e.mockapi.io/NFTS

Not sure which will work ?or am i wrong about this ?

**Comment by treeder** [comment]
Just return one at a time, like in the issue description.

**Reply by wwitkdqsck** [reply]
okay @treeder @r-gochain ill try again! thanks for the guidance !

**Reply by wwitkdqsck** [reply]
Can i add a another field "id:" in the JSON file then use this as the API Url ->
https://61f4e42262f1e300173c3f3e.mockapi.io/NFTS/{id=?}

**Reply by treeder** [reply]
You can do whatever you want, it's your API. Zeromint just makes a request to whatever URL you provide.

**Comment by wwitkdqsck** [comment]
👍 Okay @treeder noted on that! , I'll keep working on it, one more thing..
So the url is just to provide one data at a time, if it's the same.. Will
Zeromint will fetch it again?

**Reply by wwitkdqsck** [reply]
okay @treeder i've already updated the JSON file with id=0,1,....to 9

https://61f4e42262f1e300173c3f3e.mockapi.io/NFTS/{id=?}
So will this work now?
Or 
Actually ur asking something like this  https://wwitkdqsck.github.io/coloursnft-api/ to return 1 data at a time ?

**Reply by treeder** [reply]
Doesn't look like it works. I don't think mockapi will work very well for this if you can't program it to do what you want.

**Reply by wwitkdqsck** [reply]
So i need to do like the second option listed above without using MockAPI ?thats returns 1 data at a time?
